source/fe/fe_dgp.cc

```diff
@@ -25,21 +25,30 @@ DEAL_II_NAMESPACE_OPEN
 
 template <int dim, int spacedim>
 FE_DGP<dim, spacedim>::FE_DGP(const unsigned int degree)
+  : FE_DGP<dim, spacedim>(
+      Polynomials::Legendre::generate_complete_basis(degree))
+{}
+
+template <int dim, int spacedim>
+FE_DGP<dim, spacedim>::FE_DGP(
+  const std::vector<Polynomials::Polynomial<double>> &poly)
   : FE_Poly<dim, spacedim>(
-      PolynomialSpace<dim>(
-        Polynomials::Legendre::generate_complete_basis(degree)),
-      FiniteElementData<dim>(get_dpo_vector(degree),
+      PolynomialSpace<dim>(poly),
+      FiniteElementData<dim>(get_dpo_vector(poly.size() - 1),
                              1,
-                             degree,
+                             poly.size() - 1,
                              FiniteElementData<dim>::L2),
```

Comment:
This is one of the points I tried to make in the comment above: We derive the degree from the size of the polynomials. Which clearly makes sense for a sane polynomial space. But by giving the space into the hand of the user, our internal variable names get very blurry.

Reply:
I think that could probably be addressed by adding an assertion to the body of constructor, right? Right now, you already assert that for each given polynomial, `p.degree() <= degree`. You will also need to assert that the max of the `p.degree()` is == `degree`.